### PR TITLE
Update to Rust 2018 edition

### PR DESCRIPTION
**Pull request by lzutao**
By running `cargo fix --edition --edition-idioms`.

### CODE REVIEW

**Comment by philipc** [comment]
Should you do 'cargo fix --edition-idioms' too ?

**Comment by lzutao** [comment]
Yes, I did.

**Comment by philipc** [comment]
It still has the `extern crate` in lib.rs.

**Comment by lzutao** [comment]
As far as I know, `cargo fix` does not do that for you. There are plans for automatically converting `extern crate` to just use `use` statement. But it's not there yet.

**Comment by philipc** [comment]
Please try running `cargo fix --edition-idioms` again. Or at least enable `#![warn(unused_extern_crates)]` and run cargo fix.

**Comment by lzutao** [comment]
I tried them. No differences found.

**Comment by lzutao** [comment]
@philipc I've manually convert `extern crate` to `use` statement.

File: examples/nm.rs [hunk 1]
```diff
@@ -1,8 +1,6 @@
-extern crate memmap;
-extern crate object;
-
 use std::{env, fs, process};
 
+use memmap;
```
Comment (bjorn3) [inline examples/nm.rs]: This is not necessary, because other crates get added to the prelude in the 2018.

File: examples/objdump.rs [hunk 2]
```diff
@@ -1,8 +1,6 @@
-extern crate memmap;
-extern crate object;
-
 use std::{env, fs, process};
 
+use memmap;
```
Comment (bjorn3) [inline examples/objdump.rs]: Same here

File: src/lib.rs [hunk 3]
```diff
@@ -22,15 +22,7 @@ extern crate alloc;
...
-#[cfg(feature = "compression")]
-extern crate flate2;
-
-extern crate goblin;
-extern crate scroll;
-extern crate uuid;
-
-#[cfg(feature = "wasm")]
-extern crate parity_wasm;
+use goblin;
```
Comment (bjorn3) [inline src/lib.rs]: Same here

File: examples/nm.rs (hunk 1 above)
Comment (lzutao) [reply examples/nm.rs]: Thanks for spotting. I didn't know that before.

File: src/wasm.rs [hunk 4]
```diff
@@ -1,10 +1,13 @@
-use alloc::vec::Vec;
+use crate::alloc::vec::Vec;
+#[cfg(feature = "wasm")]
```
Comment (bjorn3) [inline src/wasm.rs]: This file is only compiled when the wasm feature is used, so this line is not necessary.

Comment (lzutao) [reply src/wasm.rs]: You're right. Fixed.

**Comment by philipc** [comment]
> I tried them. No differences found.

You need to check your setup then. It works for me using rust 1.33.

**Comment by lzutao** [comment]
@philipc I've those commands in 2 machines and the output is the same. Don't know what happened on my side. :)

**Pull request merged**